probe-rs-tools/src/bin/probe-rs/cmd/cargo_embed/rttui/app.rs

```diff
@@ -62,18 +62,16 @@ impl App {
                 });
             }
 
-            if up_channels.insert(number, (up, None)).is_some() {
-                return Err(anyhow!("Duplicate up channel configuration: {number}"));
-            }
+            up_channels.push((up, None));
         }
-        for down in rtt.active_down_channels.into_values() {
+        for down in rtt.active_down_channels.into_iter() {
             let number = down.number();
             if !tab_config
                 .iter()
                 .any(|tab| tab.down_channel == Some(number))
             {
                 tab_config.push(TabConfig {
-                    up_channel: if up_channels.contains_key(&number) {
+                    up_channel: if up_channels.len() > number {
```

Comment:
The behavior sounds crooked now. Is it not possible to have let's say up channels `1, 2, 5, 6`?

Reply:
It is actually not :) Up channels are numbered according to their position when we create them, and we never drop a channel struct anywhere.

We can configure cargo-embed to only display 1,2,5,6 but that is different, currently we poll **all** RTT channels regardless of whether they are displayed.

Reply:
That being said, I'm happy to park this for a bit. There might be more changes I want to do (apart from a bunch of UX and log stuff for rttui), and it's pretty untested :)

Reply:
Nono, that's fine then :) The change looks very sane :)